### ISSUE TITLE
Content sanitization?

### ISSUE THREAD

**Issue by snark** [feed-rs/feed-rs#249]
I'm working on a small feed aggregator, and as a safety consideration I'd like to strip dangerous HTML from e.g. entry.content; however, thinking about it, this is probably a step best performed by the feed parser itself. Would you be willing to accept a PR adding sanitization as a library feature? (This would probably come with an optional dependency on `ammonia`, and I'm not sure about your philosophy about adding additional dependencies.)

**Comment by markpritchard** [comment]
Hey @snark , I like to keep the dependencies of feed-rs fairly small but do like the idea of adding HTML sanitisation as an optional feature.

Happy to look at a PR that adds it, and ammonia seems like a solid choice too.

Would be good to see it exposed as an option on the parser builder, then a set of sensible defaults (e.g. relative URLs are reasonable given they are resolved in the feed processing).

**Comment by snark** [comment]
I put up PR #250, which probably merits some discussion. (Happy to address any code style requests you have, whether we should expose ammonia allowlisting behavior so users can exempt particular tags, etc.) This does what it says on the tin, but the sanitization I've provided less useful than Python's `feedparser` library, which I tend to view as the gold standard for the "liberal in what you accept" part of Postel's law. (This may be out of date, and I'm happy to look at other code for inspiration.)

feedparser aggressively upcasts `text/plain` items to `text/html` if they look "html-ish" (they contain tags and parse as HTML, basically), on the theory that a lot of people are using text fields incorrectly (see https://github.com/kurtmckee/feedparser/blob/94897c256969604613f5e7774f3c2b5d8cc1fdcc/feedparser/mixin.py#L545).  When you combine feed-rs's use of `text/plain` as the default (which I think is _correct_!) with the (I think also correct!) decision not to escape plain text, there may be surprising behavior for users of the library. (This should not be the case for Atom or JSON Feed, which do a good job of requiring feed producers to declare how their content should be interpreted.) I'm not sure how you would feel about a Parser option to do this in (mostly RSS2) feeds.

The other notable thing is that `ammonia` adds `rel="noopener no referrer"` to `a` tags; this is I think fine but may be surprising to people. We could turn it off in the new Text and Context `sanitize()` functions (or make it a Parser option, etc.)

There's some probably other stuff in the feedparser's sanitization behavior that this won't catch. For instance, I saw that feedparser decodes base64-encoded content, and then sanitizes it, but I thought that behavior would be out of scope for this change. At some point if you're interested we can probably get to parity and try to test against all the feedparser sanitizer test cases, but this feels like a reasonable first step.
